### ISSUE TITLE
using diss as a library

### ISSUE THREAD

**Issue by errordeveloper** [yazgoo/rdtach#15]
It would be great if diss had a low-level library with just the basic functionality, and it would be up to the user to control threading and forking of server process etc.
For my use case, I'd rather avoid forking and demonising the server. And perhaps I'd use lightweight threading.

**Comment by errordeveloper** [comment]
@yazgoo I would imagine you have no time for this kind of refactoring but do let me know if you like the idea either way. If I use diss, I'd have to embark on this kind of PR, or otherwise, it would have to be a rewrite.

**Comment by yazgoo** [comment]
I agree that's a good idea, I'd also like to improve the way the lib handle parallelism (right now it's using blocking IOs with timeouts, I think it could do better).
